control-plane/agents/src/bin/core/controller/io_engine/v0/nexus.rs

```diff
@@ -240,6 +240,26 @@ impl crate::controller::io_engine::NexusChildActionApi<NexusChildActionContext>
     }
 }
 
+#[async_trait::async_trait]
+impl crate::controller::io_engine::SnapshotApi for super::RpcClient {
+    async fn create_nexus_snapshot(
+        &self,
+        _request: &CreateNexusSnapshot,
+    ) -> Result<SnapshotId, SvcError> {
+        // will not be implemented for version v0.
+        unimplemented!()
+    }
+
+    /// Create a replica snapshot via gRPC.
+    async fn create_replica_snapshot(
```

Comment:
This doesn't quite fit here since this is the nexus module. I think this is addressed by a generic snapshot trait as suggested above.

Reply:
hmm probably not, you might have to add them and return grpc status not implemented.